impeller/entity/entity_pass.cc

```diff
@@ -62,12 +62,30 @@ void EntityPass::SetDelegate(std::shared_ptr<EntityPassDelegate> delegate) {
 
 void EntityPass::SetBoundsLimit(std::optional<Rect> bounds_limit) {
   bounds_limit_ = bounds_limit;
+  bounds_might_clip_ = bounds_limit.has_value();
+  bounds_are_snug_ = false;
 }
 
 std::optional<Rect> EntityPass::GetBoundsLimit() const {
   return bounds_limit_;
 }
 
+void EntityPass::SetBoundsMightClipContent(bool clips) {
```

Comment:
I have a preference for keeping the state as an enum instead of creating two different bools.

Reply:
Maybe I can reuse the same enum that Aiks Canvas uses.

Reply:
Consolidated Canvas and EntityPass into a single `ContentBoundsPromise` enum for consistency in the last commit.